### PR TITLE
[wip] Test for deals expiring in batchers

### PR DESCRIPTION
**Pull request by magik6k**
This will address https://github.com/filecoin-project/lotus/issues/6157

Draft because the test is not triggering the expected logic yet / has some incorrect asserts, but already shows that the thing works at least ~80% as expected

### CODE REVIEW

File: extern/storage-sealing/precommit_batch.go [hunk 1]
```diff
@@ -171,8 +171,12 @@ func (b *PreCommitBatcher) maybeStartBatch(notif, after bool) ([]sealiface.PreCo
...
 	}
 
-	if notif && total < cfg.MaxPreCommitBatch {
-		return nil, nil
+	if notif {
+		if total < cfg.MaxPreCommitBatch {
+			return nil, nil
+		}
+
+		log.Info("processing batch: queued more than Max")
```
Comment (arajasek) [inline extern/storage-sealing/precommit_batch.go:179]: can this not happen at exactly Max?

**Comment by arajasek** [comment]
Nice work!

**Comment by arajasek** [comment]
@gagizoned I don't think this should be in the milestone, but def good to get this across the finish line

**Comment by magik6k** [comment]
After fighting with this test some more I'm pretty sure that this specific case can't actually happen, which is why it's basically impossible to make this test do what it says in the title.

**Pull request closed without merging**